mne/epochs.py

```diff
@@ -1342,6 +1342,8 @@ def _get_data(self, out=True, verbose=None):
                         epoch = self._data[idx]
                 else:  # from disk
                     epoch_noproj = self._get_epoch_from_raw(idx)
+                    if epoch_noproj is None:  # Dropped due to bad segment
+                        continue
```

Comment:
But looking into it deeper revealed that it was not as simple as it seemed. I'd rather keep it as it is. This only makes sure that there isn't ``'NO_DATA'`` added to drop log every time it got rejected by annotations.